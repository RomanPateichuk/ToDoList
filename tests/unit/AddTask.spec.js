import { mount } from "@vue/test-utils";
import AddTaskComponent from "../../src/components/atoms/AddTask.vue";

describe("Тестирование компонента AddTask:", () => {
  const wrapper = mount(AddTaskComponent, {
    propsData: {},
  });
  const addNewTask = jest.spyOn(wrapper.vm, "addNewTask");
  const input = wrapper.findComponent("input[type='text']");

  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Должен содержать input с type='text'", () => {
    expect(input.exists()).toBe(true);
  });

  test("Должен вызывать обработчик 'callAddTask' с параметром при нажатии enter", async () => {
    await input.setValue("новый таск");
    await input.trigger("keydown.enter");
    expect(wrapper.emitted().callAddTask).toBeTruthy();
    expect(wrapper.emitted().callAddTask[0][0]).toBe("новый таск");
  });

  test("Должен вызывать метод 'addNewTask' при нажатии enter", async () => {
    await input.trigger("keydown.enter");
    expect(addNewTask).toHaveBeenCalled();
    expect(addNewTask).toHaveBeenCalledTimes(2);
  });

  test("Должен очищать input после ввода", async () => {
    input.element.value = "новый таск";
    input.trigger("input");
    await input.trigger("keydown.enter");
    expect(input.element.value).toBe("");
  });
});
