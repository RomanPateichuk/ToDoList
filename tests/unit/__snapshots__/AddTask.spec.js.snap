// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента AddTask: соответствие снимку 1`] = `<div><input type="text" placeholder="Add a new task"></div>`;
